### PR TITLE
app: forward/back buttons in titlebar

### PR DESCRIPTION
**Pull request by limitedmage**
Closes https://github.com/sourcegraph/sourcegraph/issues/51129

Shows back/forward navigation buttons in titlebar for the Tauri app

- Back button is always shown but disabled if can't navigate back
- Forward button is only shown if usable

Note: due to our recent upgrade to React Router v6, I had to manually keep track of history. This is a bit hacky, unfortunately :( I think I got most edge cases handled, but I'm not 100% certain.

CleanShot 2023-04-25 at 15 37 35@2x

## Test plan

Test manually on both main app features (search, repo pages) and opening/closing setup wizard

🚨 I was not able to test this on Linux, I don't know if it works.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.06% (+1.64 kb) | 0.01% (+1.64 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jp/appbackforward/compare-report.html;tab=live_object) for a full comparison between the commits 2a7d24c and 54d5ca0 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by vovakulikov — APPROVED** [review]
Approved, just want to say that I'm not surprised that some API that we want to have in the react routed doesn't exist (this is just sad)

File: client/web/src/app/TauriTitleBar.module.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    top: 0;
+
+    height: 32px; // This size worked for me on MacOS
+    flex-shrink: 0;
+    width: 100%;
+    user-select: none;
+    cursor: default;
+    background-color: var(--color-bg-1); // Save as global navbar
+
+    padding-left: 72px; // Eyeball for MacOS native decorations
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/app/TauriTitleBar.module.scss:15]: The user agent should provide that. But the user agent won't tell us, eg. what linux window manager [theme](https://itsfoss.com/best-gtk-themes/) you're using, which can put the decorations in weird positions. I don't know if Tauri even supports overlay-style titlebars on Linux (or Windows for that matter).

[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/app/TauriTitleBar.module.scss:15]: FWIW Electron doesn't support titlebar overlays on Linux, only Mac and Windows.

**Review by almeidapaulooliveira — APPROVED** [review]

**Pull request merged**